pkg/bwagreement/server.go

```diff
@@ -100,6 +107,10 @@ func (s *Server) BandwidthAgreements(stream pb.Bandwidth_BandwidthAgreementsServ
 		case <-ctx.Done():
 			return nil
 		case agreement := <-ch:
+			err = s.verifySignature(ctx, agreement)
+			if err != nil {
+				return err
```

Comment:
It might be better to just log this error and `continue`

Reply:
Done